pkg/terraform/executor_test.go

```diff
@@ -1,55 +1,100 @@
-//+build e2e
+// Copyright 2020 The Lokomotive Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
 
-package terraform
+package terraform_test
 
 import (
+	"fmt"
 	"io/ioutil"
 	"os"
-	"strings"
+	"path/filepath"
 	"testing"
+
+	"github.com/kinvolk/lokomotive/pkg/terraform"
 )
 
-func executor(t *testing.T) *Executor {
-	tmpDir, err := ioutil.TempDir("", "lokoctl-tests-")
-	if err != nil {
-		t.Fatalf("Creating tmp dir should succeed, got: %v", err)
+//nolint:funlen
+func TestVersionConstraint(t *testing.T) {
+	cases := map[string]struct {
+		output      string
+		expectError bool
+	}{
+		"valid": {
+			output: "Terraform v0.12.10",
+		},
+		"outdated": {
+			output:      "Terraform v0.11.0",
+			expectError: true,
+		},
+		"unsupported": {
+			output:      "Terraform v0.13.5",
+			expectError: true,
+		},
+		"with extra test": {
+			output: `Terraform v0.12.11
+
+Your version of Terraform is out of date! The latest version
```

Comment:
I think we should not write exact latest version here, as this is subject to change. Maybe we can write something like this

```suggestion
Your version of Terraform is out of date! You can update by downloading from https://www.terraform.io/downloads.html`, 
```

Reply:
This is output copied from Terraform binary as a test case, I don't think we should be modifying it. It will be removed anyway in #824.